packages/algob/src/internal/util/deployer.ts

```diff
@@ -0,0 +1,69 @@
+import { mkAccountIndex } from "../../lib/account";
+import { AlgoOperator } from "../../lib/algo-operator";
+import { loadASAFile } from "../../lib/asa";
+import { loadCheckpointsRecursive } from "../../lib/script-checkpoints";
+import type {
+  Accounts,
+  AlgobDeployer,
+  AlgobRuntimeEnv,
+  ASADefs,
+  CheckpointRepo
+} from "../../types";
+import {
+  DeployerDeployMode,
+  DeployerRunMode
+} from "../deployer";
+import { txWriter, TxWriterImpl } from "../tx-log-writer";
+
+export interface DeployerConfig {
+  runtimeEnv: AlgobRuntimeEnv
+  cpData: CheckpointRepo
+  asaDefs: ASADefs
+  algoOp: AlgoOperator
+  accounts: Accounts
+  txWriter: txWriter
+}
+
+export function mkDeployer (
+  allowWrite: boolean,
+  deployerConfig: DeployerConfig
+): AlgobDeployer {
+  if (allowWrite) {
+    return new DeployerDeployMode(
+      deployerConfig.runtimeEnv,
+      deployerConfig.cpData,
+      deployerConfig.asaDefs,
+      deployerConfig.algoOp,
+      deployerConfig.accounts,
+      deployerConfig.txWriter);
+  }
```

Comment:
can we pass it as `deployerConfig` ? and use `this.runtimeEnv = deployerConfig.runtimeEnv` in deployer constructor.
this will reduce the repetitve code.
also i suggest to rename it as `deployerCfg`.

Reply:
Ok let's change to `deployerCfg`. 
Yes we can pass the whole object to deployer and destructure it in the deployer constructer. With that the test would also need to be updated and in some of them, `asaDefs` are passed separately (some other value). So to accomodate that we would have to pass `cpData` explicitely to `MkDeploy...` and also create it before passing. I think the scope of this PR was to clean up the construction by inroducing an object. Please let me know if this makes sense. 
I will update it if you say otherwise.

Reply:
i think that's why we had `fakeDeployer` , so we don't have to do changes everywhere. but that was removed

Reply:
makes sense

Reply:
You can also overwrite the `asaDef` later on (if that helps): `deployerCfg =...` .
I also like the idea to passing the `deployerCfg` to the `DeployerDeployMode` constructor.